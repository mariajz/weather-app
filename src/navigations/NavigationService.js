import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import React from 'react';
import { BackgroundImage } from '../commons/visual-elements';
import Dashboard from '../screens/dashboard';
import ContentLoader from '../screens/content-loader';

const NavigationService = () => {
    const Stack = createNativeStackNavigator();
    const navTheme = {
        colors: {
            background: 'transparent',
        },
    };
    return (
        <NavigationContainer theme={navTheme}>
            <BackgroundImage />
            <Stack.Navigator
                screenOptions={{
                    headerShown: false,
                }}
                initialRouteName="ContentLoaderScreen">
                <Stack.Screen
                    name="ContentLoaderScreen"
                    component={ContentLoader}
                />
                <Stack.Screen name="Dashboard" component={Dashboard} />
            </Stack.Navigator>
        </NavigationContainer>
    );
};

export default NavigationService;
