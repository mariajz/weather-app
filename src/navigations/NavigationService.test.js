import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import NavigationService from './NavigationService';

jest.mock('../commons/visual-elements', () => ({
    BackgroundImage: () => <MockView>BackgroundImage</MockView>,
}));
jest.mock('../screens/dashboard', () => () => <MockView>Dashboard</MockView>);
jest.mock('../screens/permissions', () => () => (
    <MockView>Permissions</MockView>
));
jest.mock('../screens/content-loader', () => () => (
    <MockView>ContentLoader</MockView>
));

const MockNavigator = props => {
    const { children, ...restProps } = props;
    return <MockView {...restProps}>{children}</MockView>;
};
const MockScreen = props => {
    const { children, ...restProps } = props;
    return <MockView {...restProps}>{children}</MockView>;
};
jest.mock('@react-navigation/native-stack', () => ({
    createNativeStackNavigator: () => ({
        Navigator: props => <MockNavigator {...props} />,
        Screen: props => <MockScreen {...props} />,
    }),
}));
describe('NavigationService', () => {
    it('should render NavigationService', () => {
        const container = render(<NavigationService />);

        expect(container).toMatchSnapshot();
    });
});
