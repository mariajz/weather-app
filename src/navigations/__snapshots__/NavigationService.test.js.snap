// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationService should render NavigationService 1`] = `
[
  <View>
    BackgroundImage
  </View>,
  <View
    initialRouteName="ContentLoaderScreen"
    screenOptions={
      {
        "headerShown": false,
      }
    }
  >
    <View
      component={[Function]}
      name="ContentLoaderScreen"
    />
    <View
      component={[Function]}
      name="Dashboard"
    />
  </View>,
]
`;
